File: frontend/src/types/tienda.ts
export type EstadoPedidoTienda = 'pendiente' | 'parcial' | 'pagado'

export interface PedidoTienda {
  id: string
  nombre_tienda: string
  fecha_pedido: string | null
  monto_total: number
  monto_30: number
  monto_70: number
  monto_comision: number
  fecha_pago_30: string | null
  fecha_estimada_entrega: string | null
  fecha_real_entrega: string | null
  fecha_estimada_pago_70: string | null
  fecha_pago_70: string | null
  pct_comision_tienda: number
  fecha_comision: string | null
  empleada_id: string | null
  empleada_nombre: string | null
  notas: string | null
  estado: EstadoPedidoTienda
  dias_para_pago_70: number | null
  alerta_pago_70: boolean
  created_at: string
}

export interface PedidoTiendaCreate {
  nombre_tienda: string
  fecha_pedido?: string | null
  monto_total: number
  fecha_pago_30?: string | null
  fecha_estimada_entrega?: string | null
  fecha_real_entrega?: string | null
  fecha_estimada_pago_70?: string | null
  fecha_pago_70?: string | null
  pct_comision_tienda?: number
  fecha_comision?: string | null
  empleada_id?: string | null
  notas?: string | null
}

export interface EmpleadaResumen {
  id: string
  nombre: string
}

export interface ComisionItem {
  pedido_id: string
  nombre_tienda: string
  pct_comision_tienda: number
  monto_comision: number
  fecha_comision: string | null
  fecha_pedido: string | null
  empleada_nombre: string | null
}

export interface ComisionesReporte {
  items: ComisionItem[]
  total_comision: number
  desde: string | null
  hasta: string | null
}

File: backend/app/schemas/tienda.py
from datetime import date, datetime

from pydantic import BaseModel


class EmpleadaResumen(BaseModel):
    """Empleada disponible para asociar a un pedido / ranking."""

    id: str
    nombre: str


class PedidoTiendaCreate(BaseModel):
    """Alta de un pedido a tienda."""

    nombre_tienda: str
    fecha_pedido: date | None = None
    monto_total: float = 0
    fecha_pago_30: date | None = None
    fecha_estimada_entrega: date | None = None
    fecha_real_entrega: date | None = None
    fecha_estimada_pago_70: date | None = None
    fecha_pago_70: date | None = None
    pct_comision_tienda: float = 0
    fecha_comision: date | None = None
    empleada_id: str | None = None
    notas: str | None = None


class PedidoTiendaUpdate(BaseModel):
    """Edición parcial de un pedido a tienda (campos omitidos no cambian)."""

    nombre_tienda: str | None = None
    fecha_pedido: date | None = None
    monto_total: float | None = None
    fecha_pago_30: date | None = None
    fecha_estimada_entrega: date | None = None
    fecha_real_entrega: date | None = None
    fecha_estimada_pago_70: date | None = None
    fecha_pago_70: date | None = None
    pct_comision_tienda: float | None = None
    fecha_comision: date | None = None
    empleada_id: str | None = None
    notas: str | None = None


class PedidoTiendaResponse(BaseModel):
    """Pedido a tienda con los montos y el estado calculados."""

    id: str
    nombre_tienda: str
    fecha_pedido: date | None
    monto_total: float
    # 30/70/comisión calculados del total (CNY)
    monto_30: float
    monto_70: float
    monto_comision: float
    fecha_pago_30: date | None
    fecha_estimada_entrega: date | None
    fecha_real_entrega: date | None
    fecha_estimada_pago_70: date | None
    fecha_pago_70: date | None
    pct_comision_tienda: float
    fecha_comision: date | None
    empleada_id: str | None
    empleada_nombre: str | None
    notas: str | None
    estado: str  # pendiente / parcial / pagado
    dias_para_pago_70: int | None
    alerta_pago_70: bool
    created_at: datetime


class ComisionItem(BaseModel):
    """Una comisión recibida (derivada de un pedido a tienda)."""

    pedido_id: str
    nombre_tienda: str
    pct_comision_tienda: float
    monto_comision: float
    fecha_comision: date | None
    fecha_pedido: date | None
    empleada_nombre: str | None


class ComisionesReporte(BaseModel):
    """Comisiones recibidas en un período con su total."""

    items: list[ComisionItem]
    total_comision: float
    desde: date | None
    hasta: date | None
